src/test/java/picard/TestDataProviders.java

```diff
@@ -21,21 +26,46 @@
 public class TestDataProviders {
 
     @Test
-    public void testAllDataProviders() throws IllegalAccessException, InstantiationException, InvocationTargetException {
-        int i = 0;
+    public void IndependentTestOfDataProviderTest() throws IllegalAccessException, InvocationTargetException, InstantiationException {
+        testAllDataProvidersdata();
+    }
+
+    @DataProvider(name = "DataprovidersThatDontTestThemselves")
+    public Iterator<Object[]> testAllDataProvidersdata() throws IllegalAccessException, InstantiationException, InvocationTargetException {
+
+        List<Object[]> data = new ArrayList<>();
         final ClassFinder classFinder = new ClassFinder();
         classFinder.find("picard", Object.class);
-        for (Class testClass : classFinder.getClasses()) {
+
+        for (final Class<?> testClass : classFinder.getClasses()) {
             if (Modifier.isAbstract(testClass.getModifiers())) continue;
             for (final Method method : testClass.getMethods()) {
                 if (method.isAnnotationPresent(DataProvider.class)) {
-                    System.err.println("Method: " + testClass.getName() + ":" + method.getName());
-                    method.invoke(testClass.newInstance());
-                    i++;
+                    data.add(new Object[]{method, testClass});
                 }
             }
         }
-        Assert.assertNotSame(i,0);
-        System.err.println("Found: "+ i + " @DataProviders.");
+        Assert.assertTrue(data.size() > 1);
```

Comment:
I think in general you don't want to put asserts inside a dataprovider. If you want to do this, you should do it inside `IndependentTestOfDataProviderTest()` above.